packages/frontend/editor-ui/src/features/ai/chatHub/ChatView.vue

```diff
@@ -423,24 +422,38 @@ async function onUpdateTools(newTools: INode[]) {
 	}
 }
 
+function closeAgentEditor() {
+	// TODO: Do we need some handling here?
+}
+
 async function handleEditAgent(agentId: string) {
 	try {
 		await chatStore.fetchCustomAgent(agentId);
-		editingAgentId.value = agentId;
-		uiStore.openModal('agentEditor');
+
+		uiStore.openModalWithData({
+			name: AGENT_EDITOR_MODAL_KEY,
+			data: {
+				agentId,
+				credentials: credentialsByProvider,
+				onClose: closeAgentEditor,
+				onCreateCustomAgent: handleSelectModel,
+			},
+		});
 	} catch (error) {
 		toast.showError(error, 'Failed to load agent');
 	}
 }
 
 function openNewAgentCreator() {
 	chatStore.currentEditingAgent = null;
-	editingAgentId.value = undefined;
-	uiStore.openModal('agentEditor');
-}
-
-function closeAgentEditor() {
-	editingAgentId.value = undefined;
+	uiStore.openModalWithData({
```

Comment:
Not to be fixed in this PR, but it seems fading animation on closing modal is lost with this approach